system/system.go

```diff
@@ -47,13 +61,185 @@ func Info() (types.System, error) {
 		system.Network = network
 	}
 
+	vmSystem, vmRole, err := host.Virtualization()
+	if err == nil {
+		system.VMSystem = vmSystem
+		system.VMRole = vmRole
+	}
+
+	if runtime.GOOS == "linux" {
+		libcType, err := getLibCType()
+		if err == nil {
+			system.LibCType = libcType
+		}
+	}
+
 	return system, nil
 }
 
+func getLibCType() (string, error) {
+	output, err := exec.Command("ldd", "--version").CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	text := strings.ToLower(string(output))
+	if strings.Contains(text, "gnu") {
+		return "glibc", nil
+	}
+	if strings.Contains(text, "musl") {
+		return "musl", nil
+	}
+	return "", nil
+}
+
+// GetCloudProvider inspects local files, looks up hostnames, and makes HTTP
+// requests in order to determine if the local system is running within a cloud
+// provider.
+func GetCloudProvider(ctx context.Context) string {
+	switch runtime.GOOS {
+	case "linux":
+		// Issue a dmidecode command to see if we are on EC2 or GCP
+		logger.Debug("sudo -n dmidecode -s bios-version")
```

Comment:
I'm assuming sudo is probably required to run this, but this could pose an issue for agents that don't have that access. Maybe we should just make a point in the docs about this.. wdyt @echlebek cc @hillaryfraley

Reply:
This is only the first attempt. If sudo is not available, then the execution will fail, and a fallback mechanism will be used.

Reply:
Yes, I see that now. Doesn't look like those fallback mechanisms require sudo so this should be fine!